packages/eslint-plugin/docs/rules/no-unnecessary-condition.mdx

```diff
@@ -15,7 +15,7 @@ Conversely, any expression that always evaluates to truthy or always evaluates t
 The following expressions are checked:
 
 - Arguments to the `&&`, `||` and `?:` (ternary) operators
-- Conditions for `if`, `for`, `while`, and `do-while` statements
+- Conditions for `if`, `switch`, `for`, `while`, and `do-while` statements
```

Comment:
(nit) I think "conditions for `switch` statements" isn't _quite_ as self-explanatory as the rest of these. What about the following? Or some other way of being explicit that it's the `case`s that are checked?

```suggestion
- Conditions for `if`, `for`, `while`, and `do-while` statements
- `case`s in `switch` statements
```